Mathlib/Topology/KrullDimension.lean

```diff
@@ -0,0 +1,24 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Fangming Li
+-/
+
+import Mathlib.Topology.Irreducible
+import Mathlib.Order.KrullDimension
+
+/-!
+# The Krull dimension of a topological space
+
+The Krull dimension of a topological space is the order theoretic Krull dimension applied to the
+collection of all its subsets that are closed and irreducible. Unfolding this definition, it is
+the length of longest series of closed irreducible subsets ordered by inclusion.
+-/
+
+/--
+The Krull dimension of a topological space is the supremum of lengths of chains of
+closed irreducible sets.
+-/
+noncomputable def topologicalKrullDim (T : Type _) [TopologicalSpace T] :
+    WithBot (WithTop ℕ) :=
+  krullDim { s : Set T | IsClosed s ∧ IsIrreducible s }
```

Comment:
I think yes. `irreducibleComponents_eq_maximals_closed` also mentions this.

Reply:
So does `vanishingIdeal_isClosed_isIrreducible` (this time in the other order). This proof is only used once in mathlib, in the proof of `vanishingIdeal_irreducibleComponents` (which is now worse than it was). To avoid this regression probably `vanishingIdeal_isClosed_isIrreducible` should be renamed `vanishingIdeal_isIrreducibleClosed` and fixed up to use this new concept.

Reply:
> So does `vanishingIdeal_isClosed_isIrreducible` (this time in the other order). This proof is only used once in mathlib, in the proof of `vanishingIdeal_irreducibleComponents` (which is now worse than it was). To avoid this regression probably `vanishingIdeal_isClosed_isIrreducible` should be renamed `vanishingIdeal_isIrreducibleClosed` and fixed up to use this new concept.

I have made changes according to your advice. Consequently, the names of some results have been changed.

Reply:
I have added a structure `IrreducibleCloseds` and refactored some declarations to use it.